### DISCUSSION TITLE
How can I hide bars with null values in C3?

### DISCUSSION THREAD

**Discussion by k-yash** [d3/d3#3612] [Q&A] [answered]
I have a C3 chart with data as given in columns. 
Notice that there are null values in data1 and in between data 3. C3 is trying to display my bar chart in groupings of 5, creating a separate bar for null values. I'm trying to hide these null values.

```
var chart = c3.generate({
    data: {
        x : 'x',
        line:{connect_null: true},
        columns: [
            ['x','x1' ,'x2', 'x3', 'x4', 'x5', 'x6', 'x7'],
            ['data1',null ,30, 200, 100, 400, 150, 250],
            ['data2',null ,130, 100, 140, 200, 150, 50],
            ['data3',null ,130, null, 200, 300, -200, 100],
            ['data4',null ,20, -90, 200, 130, -20, 90],
            ['data5',-90, 20, -90, 200, 130, -20, 90]
        ],
        type: 'bar'
    },
    axis: {
        x: {
            type: 'category', // this needed to load string x value
        }
    },
    bar: {
        width: {
            ratio: 0.7 // this makes bar width 50% of length between ticks
        },
        space:0.05
        // or
        //width: 100 // this makes bar width 100px
    }
});
```
![example1](https://user-images.githubusercontent.com/59231788/207302608-9bb35909-a718-403c-8b97-2f3893a7a4e0.PNG)

**Comment by Fil** [comment, marked as answer]
Hi, this is D3, not C3.

From C3's homepage:
> For general C3.js-related discussion, please visit our [Google Group at https://groups.google.com/forum/#!forum/c3js](https://groups.google.com/forum/#!forum/c3js).
